provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/KotlinSettingsScriptIntegrationTest.kt

```diff
@@ -0,0 +1,34 @@
+package org.gradle.kotlin.dsl.integration
+
+import org.gradle.kotlin.dsl.fixtures.AbstractIntegrationTest
+
+import org.junit.Test
+
+
+class KotlinSettingsScriptIntegrationTest : AbstractIntegrationTest() {
+
+    @Test
+    fun `Settings script path is resolved relative to parent script dir`() {
+
+        withFile("gradle/my.settings.gradle.kts", """
+            apply { from("./answer.settings.gradle.kts") }
+        """)
+
+        withFile("gradle/answer.settings.gradle.kts", """
+            gradle.rootProject {
+                val answer by extra { "42" }
+            }
+        """)
+
+        withSettings("""
+            apply { from("gradle/my.settings.gradle.kts") }
+        """)
+
+        withBuildScript("""
+            val answer: String by extra
+            println("*" + answer + "*")
+        """)
+
+        assert(build().output.contains("*42*"))
```

Comment:
:o Are you sure you want to use `assert` instead of something like Hamkrest's `assertThat("blah", contains("*42*"))`
[Link.](https://github.com/npryce/hamkrest/blob/9b66019c4c84e1aee1f2966b5f47f2d5f909f8d1/src/main/kotlin/com/natpryce/hamkrest/string_matchers.kt#L108)

Reply:
👍